### PR TITLE
UIIN-1127 Drag and drop accessibility

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose
https://issues.folio.org/browse/UIIN-1127

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/Instance/HoldingsList/Holding/Holding.js [hunk 1]
```diff
@@ -53,9 +76,72 @@ const Holding = ({
...
+
   const holdingButtonsGroup = useMemo(() => {
     return (
       <>
+        {draggable && (
+          <Dropdown
+            open={isDropdownOpen}
+            onToggle={onDropdownToggle}
+          >
+            <DropdownButton
```
Comment (NikitaSedyx) [inline src/Instance/HoldingsList/Holding/Holding.js:91]: it''s deprecated and should be replaced wih renderTrigger and renderMenu

File: src/Instance/HoldingsList/Holding/Holding.js [hunk 2]
```diff
@@ -53,9 +76,72 @@ const Holding = ({
...
+  const onDropdownToggle = useCallback(() => {
+    toggleDropDown(!isDropdownOpen);
+  }, [isDropdownOpen]);
+
   const holdingButtonsGroup = useMemo(() => {
     return (
       <>
+        {draggable && (
+          <Dropdown
+            open={isDropdownOpen}
```
Comment (NikitaSedyx) [inline src/Instance/HoldingsList/Holding/Holding.js:88]: these props are not required

File: src/Instance/HoldingsList/Holding/Holding.js [hunk 3]
```diff
@@ -53,9 +76,72 @@ const Holding = ({
...
 
+  const onDropdownToggle = useCallback(() => {
+    toggleDropDown(!isDropdownOpen);
+  }, [isDropdownOpen]);
+
   const holdingButtonsGroup = useMemo(() => {
     return (
       <>
+        {draggable && (
+          <Dropdown
```
Comment (NikitaSedyx) [inline src/Instance/HoldingsList/Holding/Holding.js:87]: please create separate component

File: src/Instance/InstanceMovement/HoldingMovementList/HoldingsListMovement.js [hunk 4]
```diff
@@ -38,31 +38,32 @@ const HoldingsListMovement = ({
...
     <>
       {
         holdings.length ? (
-          holdings.map((holding, index) => (
-            <HoldingContainer
-              key={`items_${holding.id}`}
-              instance={instance}
-              holding={holding}
-              referenceData={referenceData}
+          holdings.map((holding, index) => {
```
Comment (NikitaSedyx) [inline src/Instance/InstanceMovement/HoldingMovementList/HoldingsListMovement.js:45]: are there any changes here?

File: src/Instance/MoveItemsContext/MoveItemsContext.js [hunk 5]
```diff
@@ -50,6 +50,29 @@ const MoveItemsContext = ({ children, moveItems }) => {
     setActiveDropZone(undefined);
   }, [selectedItemsMap]);
 
+  const onSelect = useCallback(({ target }) => {
```
Comment (NikitaSedyx) [inline src/Instance/MoveItemsContext/MoveItemsContext.js:53]: it's 100% copy paste of `onDragEnd`

File: src/Instance/HoldingsList/Holding/MoveToDropdown.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+const MoveToDropdown = ({
+  movetoHoldings,
+  instances,
+  selectedItems,
+  onSelect,
+  draggable,
+  holding,
+}) => {
+  const dropdownButton = ({ getTriggerProps }) => (
```
Comment (NikitaSedyx) [inline src/Instance/HoldingsList/Holding/MoveToDropdown.js:20]: render with callback

File: src/Instance/MoveItemsContext/MoveItemsContext.js [hunk 7]
```diff
@@ -85,6 +85,8 @@ const MoveItemsContext = ({ children, moveItems }) => {
           selectItemsForDrag,
           isItemsDragSelected,
           getDraggingItems,
+          selectedItemsMap,
+          onSelect : onDragEnd,
```
Comment (NikitaSedyx) [inline src/Instance/MoveItemsContext/MoveItemsContext.js:89]: select i not drag, drag and select should invoke common code and not mix everything in one function

File: src/Instance/MoveItemsContext/MoveItemsContext.js [hunk 8]
```diff
@@ -50,6 +45,21 @@ const MoveItemsContext = ({ children, moveItems }) => {
     setActiveDropZone(undefined);
   }, [selectedItemsMap]);
 
+  const onSelect = useCallback(({ target }) => {
+    const from = target.dataset.itemId;
+    const to = target.dataset.toId;
+
+    itemsMovement(from, to, target);
```
Comment (NikitaSedyx) [inline src/Instance/MoveItemsContext/MoveItemsContext.js:52]: `moveItems`?

File: src/Instance/MoveItemsContext/MoveItemsContext.js [hunk 9]
```diff
@@ -50,6 +45,21 @@ const MoveItemsContext = ({ children, moveItems }) => {
...
+  }, [selectedItemsMap]);
```
Comment (NikitaSedyx) [inline src/Instance/MoveItemsContext/MoveItemsContext.js:53]: missed itemsMovement - moveItems dependency?

File: src/Instance/MoveItemsContext/MoveItemsContext.js [hunk 10]
```diff
@@ -50,6 +45,21 @@ const MoveItemsContext = ({ children, moveItems }) => {
...
+
+  const onDragEnd = useCallback((result) => {
+    if (!result.destination) return;
+    const from = result.source.droppableId;
+    const to = result.destination.droppableId;
+
+    itemsMovement(from, to, result);
+  }, [selectedItemsMap]);
```
Comment (NikitaSedyx) [inline src/Instance/MoveItemsContext/MoveItemsContext.js:61]: same?

File: src/Instance/MoveItemsContext/MoveItemsContext.js [hunk 11]
```diff
@@ -50,6 +45,21 @@ const MoveItemsContext = ({ children, moveItems }) => {
...
 
+  const onSelect = useCallback(({ target }) => {
+    const from = target.dataset.itemId;
+    const to = target.dataset.toId;
+
+    itemsMovement(from, to, target);
+  }, [selectedItemsMap]);
+
+  const onDragEnd = useCallback((result) => {
+    if (!result.destination) return;
```
Comment (NikitaSedyx) [inline src/Instance/MoveItemsContext/MoveItemsContext.js:56]: empty line here

File: src/Instance/MoveItemsContext/MoveItemsContext.js (hunk 10 above)
Comment (NikitaSedyx) [reply src/Instance/MoveItemsContext/MoveItemsContext.js:61]: do you need `selectedItemsMap` here?

File: src/Instance/MoveItemsContext/MoveItemsContext.js (hunk 9 above)
Comment (NikitaSedyx) [reply src/Instance/MoveItemsContext/MoveItemsContext.js:53]: same

**Review by aliaksei-chumakou — APPROVED** [review]

File: src/Instance/HoldingsList/Holding/Holding.js [hunk 12]
```diff
@@ -21,30 +21,38 @@ import {
...
 }) => {
-  const { locationsById } = referenceData;
-  const labelLocation = holding.permanentLocationId ? locationsById[holding.permanentLocationId].name : '';
   const {
     selectItemsForDrag,
     isItemsDragSelected,
     getDraggingItems,
     activeDropZone,
     isItemsDropable,
+    instances,
```
Comment (NikitaSedyx) [inline src/Instance/HoldingsList/Holding/Holding.js:46]: is it possible to move data related only to `MoveToDropdown` there  via context

File: src/Instance/HoldingsList/Holding/MoveToDropdown.css [hunk 13]
```diff
@@ -0,0 +1,4 @@
+.dropDownItem {
```
Comment (NikitaSedyx) [inline src/Instance/HoldingsList/Holding/MoveToDropdown.css:1]: create a folder for `MoveToDropdown`

File: src/Instance/HoldingsList/Holding/MoveToDropdown.js [hunk 14]
```diff
@@ -0,0 +1,130 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
```
Comment (NikitaSedyx) [inline src/Instance/HoldingsList/Holding/MoveToDropdown.js:4]: extra line

File: src/Instance/MoveItemsContext/MoveItemsContext.js (hunk 8 above)
Comment (uladislausamets) [reply src/Instance/MoveItemsContext/MoveItemsContext.js:52]: moveItems is a function from props

**Comment by NikitaSedyx** [comment]
have you tested permissions? looks like no

**Review by AndreiShumski2 — APPROVED** [review]

File: src/Instance/MoveItemsContext/MoveItemsContext.js (hunk 8 above)
Comment (NikitaSedyx) [reply src/Instance/MoveItemsContext/MoveItemsContext.js:52]: use another name, `itemsMovement` doesn't sound like a function

File: src/Instance/MoveItemsContext/MoveItemsContext.js (hunk 10 above)
Comment (NikitaSedyx) [reply src/Instance/MoveItemsContext/MoveItemsContext.js:61]: will you fix dependencies?

**Review by NikitaSedyx — APPROVED** [review]

**Review by ViktorSoroka07 — APPROVED** [review]

**Pull request merged**